### ISSUE TITLE
Data race in tests that change logger

### ISSUE THREAD

**Issue by martin-sucha** [gocql/gocql#1281]
Please answer these questions before submitting your issue. Thanks!

### What version of Cassandra are you using?
Go: 1.11 CASS=3.0.17 AUTH=false

### What version of Gocql are you using?
59996b52a6da045fb595758fbcbedd8219bd0914

### What did you do?
There was a data race in tests in Travis CI: https://travis-ci.org/gocql/gocql/jobs/508332390

It looks to be caused by changing global `Logger` variable. The test is accessing the logger like this:

```go
log := &testLogger{}
Logger = log
defer func() {
	Logger = &defaultLogger{}
}()
```

The following tests do this and are
susceptible to the race:

- TestDNSLookupConnected
- TestDNSLookupError
- TestQueryMultinodeWithMetrics
- TestSpeculativeExecution
- TestStartupTimeout

### What did you expect to see?
Tests pass without data race

### What did you see instead?

```
=== RUN   TestDNSLookupConnected
==================
WARNING: DATA RACE
Write at 0x000000b7a8c0 by goroutine 90:
  github.com/gocql/gocql.TestDNSLookupConnected()
      /home/travis/gopath/src/github.com/gocql/gocql/conn_test.go:152 +0xb5
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.linux.amd64/src/testing/testing.go:827 +0x162
Previous read at 0x000000b7a8c0 by goroutine 85:
  github.com/gocql/gocql.(*hostConnPool).logConnectErr()
      /home/travis/gopath/src/github.com/gocql/gocql/connectionpool.go:454 +0x124
  github.com/gocql/gocql.(*hostConnPool).connectMany.func1()
      /home/travis/gopath/src/github.com/gocql/gocql/connectionpool.go:487 +0x88
Goroutine 90 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.11.linux.amd64/src/testing/testing.go:878 +0x650
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.11.linux.amd64/src/testing/testing.go:1119 +0xa8
  testing.tRunner()
      /home/travis/.gimme/versions/go1.11.linux.amd64/src/testing/testing.go:827 +0x162
  testing.runTests()
      /home/travis/.gimme/versions/go1.11.linux.amd64/src/testing/testing.go:1117 +0x4ee
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.11.linux.amd64/src/testing/testing.go:1034 +0x2ee
  main.main()
      _testmain.go:248 +0x221
Goroutine 85 (finished) created at:
  github.com/gocql/gocql.(*hostConnPool).connectMany()
      /home/travis/gopath/src/github.com/gocql/gocql/connectionpool.go:484 +0x159
  github.com/gocql/gocql.(*hostConnPool).fill.func1()
      /home/travis/gopath/src/github.com/gocql/gocql/connectionpool.go:438 +0x42
==================
--- FAIL: TestDNSLookupConnected (0.00s)
    testing.go:771: race detected during execution of test
```

**Comment by alourie** [comment]
Hm, it's strange, the code in the example you gave is specifically used to prevent this. I'll have a look.

**Comment by martin-sucha** [comment]
Should be fixed by #1553
